src/zenml/services/service_status.py

```diff
@@ -0,0 +1,75 @@
+#  Copyright (c) ZenML GmbH 2022. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+
+from typing import Optional
+
+from pydantic import BaseModel
+
+from zenml.logger import get_logger
+from zenml.utils.enum_utils import StrEnum
+
+logger = get_logger(__name__)
+
+
+class ServiceState(StrEnum):
+    """Possible states for the service and service endpoint."""
+
+    ACTIVE = "active"
+    PENDING_STARTUP = "pending_startup"
+    INACTIVE = "inactive"
```

Comment:
Dumb question: what defines an inactive service? How do we define whether something is inactive?

Reply:
If it helps, I do think these states need some refactoring too, to make them more user-friendly, so if you have any suggestions about that, I welcome them.